src/compiler/parser.ts

```diff
@@ -3386,6 +3389,7 @@ namespace ts {
          *      6) - UnaryExpression[?yield]
          *      7) ~ UnaryExpression[?yield]
          *      8) ! UnaryExpression[?yield]
+         *      9) await AwaitExpression[?yield]

```

Comment:
Prefix this with the `[+Await]` [from the production](https://tc39.github.io/ecmascript-asyncawait/#prod-AwaitExpression)